### PR TITLE
Drop support for Python 2

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/capitalone/datacompy/issues/48.

### CODE REVIEW

**Comment by fdosani** [comment]
@hugovk Thank you for this PR. Will take a look in a bit. In the mean time (if you haven't already) could you please complete the [CLA](https://docs.google.com/forms/d/19LpBBjykHPox18vrZvBbZUcK6gQTj7qv1O5hCduAZFU/viewform)

**Comment by hugovk** [comment]
Done!

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
 enum34>=1.1.6;python_version<"3.4"
```
Comment (fdosani) [inline requirements.txt]: thinking we should just remove `enum` here too. I don't think we want to be supporting older versions of 3 either. Feel like at the minimum 3.5+

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,9 @@
 language: python
 jdk: openjdk8
 dist: trusty
-sudo: false
 cache:
   pip: true
 python:
-  - "2.7"
```
Comment (fdosani) [inline .travis.yml]: Might make sense to swap this out for:

```yaml
  - "3.5"
  - "3.6"
  - "3.7"
```

Comment (hugovk) [reply .travis.yml]: Good idea.

However, Python 3.7 requires `dist: xenial` on Travis, and a quick test shows more work is needed to test against Xenial, so I'll only add 3.6 as part of this PR and 3.7 can be dealt with in another issue.

https://travis-ci.org/hugovk/datacompy/builds/591518867

Comment (fdosani) [reply .travis.yml]: Sounds good! Thank you Hugo.

**Comment by hugovk** [comment]
Updated!

**Pull request merged**